File: versioned_docs/version-1.9.x/examples/rs/examples/buckets_change.rs
use reduct_rs::{BucketSettings, ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a new client with the API URL and API token
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // Get an existing bucket
    let bucket = client.get_bucket("example-bucket").await?;

    // Change the quota size of the bucket to 5GB
    let new_settings = BucketSettings {
        quota_size: Some(5_000_000_000),
        ..BucketSettings::default()
    };
    bucket.set_settings(new_settings).await?;

    assert_eq!(bucket.settings().await?.quota_size, Some(5_000_000_000));
    Ok(())
}

File: versioned_docs/version-1.14.x/examples/cpp/src/buckets_change.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
    // Create a client with the server URL
    auto client = IClient::Build("http://127.0.0.1:8383", {
            .api_token = "my-token"
    });

    // Get an existing bucket
    auto [bucket, get_err] = client->GetBucket("example-bucket");
    assert(get_err == Error::kOk);

    // Change the quota size of the bucket to 5GB
    IBucket::Settings new_settings;
    new_settings.quota_size = 5'000'000'000;
    auto change_err = bucket->UpdateSettings(new_settings);

    assert(change_err == Error::kOk);
    assert(bucket->GetSettings().result.quota_size == 5'000'000'000);
}
